internal/apex/route.go

```diff
@@ -62,3 +64,37 @@ func (ax *Apex) addChildPrefixRoute(childPrefix string) {
 		ax.logger.Infof("error adding the child prefix route: %v", err)
 	}
 }
+
+// handlePeerRoute when a peer is this handles route deletion
+func (ax *Apex) handlePeerRouteDelete(dev string, wgPeerConfig models.Peer) {
+	switch ax.os {
+	case Darwin.String():
+		// TODO: OSX route lookups
+		for _, allowedIP := range wgPeerConfig.AllowedIPs {
+			if err := DeleteRoute(allowedIP, dev); err != nil {
+				ax.logger.Debug(err)
+			}
+		}
+
+	case Linux.String():
+		for _, allowedIP := range wgPeerConfig.AllowedIPs {
+			routeExists, err := RouteExists(allowedIP)
+			if err != nil {
+				ax.logger.Info(err)
```

Comment:
I think you should add a `continue` here. You don't want to act on the result if an error occurs.

Do you really want `Info` here? Check the other logging in this function. It's a mix of Info and Debug, but all for error conditions.